nav2_mppi_controller/src/optimizer.cpp

```diff
@@ -248,32 +237,37 @@ void Optimizer::applyControlSequenceConstraints()
 {
   auto & s = settings_;
 
-  if (isHolonomic()) {
-    control_sequence_.vy = xt::clip(control_sequence_.vy, -s.constraints.vy, s.constraints.vy);
-  }
-
-  control_sequence_.vx = xt::clip(control_sequence_.vx, s.constraints.vx_min, s.constraints.vx_max);
-  control_sequence_.wz = xt::clip(control_sequence_.wz, -s.constraints.wz, s.constraints.wz);
-
   float max_delta_vx = s.model_dt * s.constraints.ax_max;
   float min_delta_vx = s.model_dt * s.constraints.ax_min;
   float max_delta_vy = s.model_dt * s.constraints.ay_max;
   float max_delta_wz = s.model_dt * s.constraints.az_max;
-  float vx_last = control_sequence_.vx(0);
-  float vy_last = control_sequence_.vy(0);
-  float wz_last = control_sequence_.wz(0);
-  for (unsigned int i = 1; i != control_sequence_.vx.shape(0); i++) {
+  float vx_last = std::min(
+    s.constraints.vx_max, std::max(control_sequence_.vx(0), s.constraints.vx_min));
+  float vy_last = std::min(
```

Comment:
Put vy in the `if` statement so its not computed for the ackermann/diff drive use cases